### ISSUE TITLE
Flakey tests in ImportNativeAppEngineStandardProjectTest

### ISSUE THREAD

**Issue by briandealwis** [GoogleCloudPlatform/google-cloud-eclipse#2569]
Travis fails occasionally in `ImportNativeAppEngineStandardProjectTest`, in both `importAppEngineStandardJava7_from1_3_1()` and ``.  Seen on Mars, Neon, and Oxygen.

On Mars:
```
-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Running com.google.cloud.tools.eclipse.integration.appengine.ImportNativeAppEngineStandardProjectTest
[Watchdog] > importAppEngineStandardJava7_from1_3_1(com.google.cloud.tools.eclipse.integration.appengine.ImportNativeAppEngineStandardProjectTest)
Window manager warning: Buggy client sent a _NET_ACTIVE_WINDOW message with a timestamp of 0 for 0x40032d (Import )
Window manager warning: meta_window_activate called by a pager with a 0 timestamp; the pager needs to be fixed.
Window manager warning: Buggy client sent a _NET_ACTIVE_WINDOW message with a timestamp of 0 for 0x400021 (Resource -)
Window manager warning: meta_window_activate called by a pager with a 0 timestamp; the pager needs to be fixed.
[Watchdog] > importAppEngineStandardJava8_from1_3_1(com.google.cloud.tools.eclipse.integration.appengine.ImportNativeAppEngineStandardProjectTest)
Tests run: 2, Failures: 0, Errors: 1, Skipped: 1, Time elapsed: 5.876 sec <<< FAILURE! - in com.google.cloud.tools.eclipse.integration.appengine.ImportNativeAppEngineStandardProjectTest
importAppEngineStandardJava7_from1_3_1(com.google.cloud.tools.eclipse.integration.appengine.ImportNativeAppEngineStandardProjectTest)  Time elapsed: 1.679 sec  <<< ERROR!
org.eclipse.swtbot.swt.finder.exceptions.AssertionFailedException: assertion failed: 
	at org.eclipse.swtbot.swt.finder.utils.internal.Assert.isTrue(Assert.java:95)
	at org.eclipse.swtbot.swt.finder.utils.internal.Assert.isTrue(Assert.java:83)
	at org.eclipse.swtbot.swt.finder.widgets.SWTBotTreeItem.select(SWTBotTreeItem.java:496)
	at org.eclipse.swtbot.swt.finder.widgets.SWTBotTreeItem.select(SWTBotTreeItem.java:523)
	at com.google.cloud.tools.eclipse.integration.appengine.SwtBotAppEngineActions.importNativeProject(SwtBotAppEngineActions.java:140)
	at com.google.cloud.tools.eclipse.integration.appengine.ImportNativeAppEngineStandardProjectTest.importAppEngineStandardJava7_from1_3_1(ImportNativeAppEngineStandardProjectTest.java:63)

importAppEngineStandardJava8_from1_3_1(com.google.cloud.tools.eclipse.integration.appengine.ImportNativeAppEngineStandardProjectTest) skipped

Results :

Tests in error: 
  ImportNativeAppEngineStandardProjectTest.importAppEngineStandardJava7_from1_3_1:63 » AssertionFailed

Tests run: 2, Failures: 0, Errors: 1, Skipped: 1
```

On Neon:
```
Running com.google.cloud.tools.eclipse.integration.appengine.ImportNativeAppEngineStandardProjectTest
[Watchdog] > importAppEngineStandardJava7_from1_3_1(com.google.cloud.tools.eclipse.integration.appengine.ImportNativeAppEngineStandardProjectTest)
Window manager warning: Buggy client sent a _NET_ACTIVE_WINDOW message with a timestamp of 0 for 0x40350d (Import )
Window manager warning: meta_window_activate called by a pager with a 0 timestamp; the pager needs to be fixed.
Window manager warning: Buggy client sent a _NET_ACTIVE_WINDOW message with a timestamp of 0 for 0x400021 (data - Res)
Window manager warning: meta_window_activate called by a pager with a 0 timestamp; the pager needs to be fixed.
[Watchdog] > importAppEngineStandardJava8_from1_3_1(com.google.cloud.tools.eclipse.integration.appengine.ImportNativeAppEngineStandardProjectTest)
Window manager warning: Buggy client sent a _NET_ACTIVE_WINDOW message with a timestamp of 0 for 0x40371d (Import )
Window manager warning: meta_window_activate called by a pager with a 0 timestamp; the pager needs to be fixed.
11:39:02.722 [main] WARN  o.e.s.s.finder.widgets.SWTBotButton - Widget is not enabled: (of type 'Button' and with mnemonic 'Next >' and with style 'SWT.PUSH')
11:39:02.725 [main] WARN  o.e.s.s.finder.widgets.SWTBotButton - Widget is not enabled: (of type 'Button' and with mnemonic 'Next >' and with style 'SWT.PUSH')
11:39:02.728 [main] WARN  o.e.s.s.finder.widgets.SWTBotButton - Widget is not enabled: (of type 'Button' and with mnemonic 'Next >' and with style 'SWT.PUSH')
11:39:02.728 [main] WARN  o.e.s.s.finder.widgets.SWTBotButton - Widget is not enabled: (of type 'Button' and with mnemonic 'Next >' and with style 'SWT.PUSH')
11:39:02.729 [main] WARN  o.e.s.s.finder.widgets.SWTBotButton - Widget is not enabled: (of type 'Button' and with mnemonic 'Next >' and with style 'SWT.PUSH')
11:39:02.837 [main] WARN  o.e.s.s.finder.widgets.SWTBotButton - Widget is not enabled: (of type 'Button' and with mnemonic 'Finish' and with style 'SWT.PUSH')
11:39:02.839 [main] WARN  o.e.s.s.finder.widgets.SWTBotButton - Widget is not enabled: (of type 'Button' and with mnemonic 'Finish' and with style 'SWT.PUSH')
11:39:02.842 [main] WARN  o.e.s.s.finder.widgets.SWTBotButton - Widget is not enabled: (of type 'Button' and with mnemonic 'Finish' and with style 'SWT.PUSH')
11:39:02.846 [main] WARN  o.e.s.s.finder.widgets.SWTBotButton - Widget is not enabled: (of type 'Button' and with mnemonic 'Finish' and with style 'SWT.PUSH')
11:39:02.848 [main] WARN  o.e.s.s.finder.widgets.SWTBotButton - Widget is not enabled: (of type 'Button' and with mnemonic 'Finish' and with style 'SWT.PUSH')
Window manager warning: Buggy client sent a _NET_ACTIVE_WINDOW message with a timestamp of 0 for 0x400021 (data - Res)
Window manager warning: meta_window_activate called by a pager with a 0 timestamp; the pager needs to be fixed.
Nov 09, 2017 11:39:04 AM com.google.cloud.tools.eclipse.appengine.libraries.persistence.LibraryClasspathContainerSerializer loadLibraryIds
INFO: Library-id state file not found for: AESv8
Window manager warning: Buggy client sent a _NET_ACTIVE_WINDOW message with a timestamp of 0 for 0x403891 (Initialize)
Window manager warning: meta_window_activate called by a pager with a 0 timestamp; the pager needs to be fixed.
11:39:05.643 [Worker-2] INFO  o.e.m.c.i.e.AbstractTransferListenerAdapter - Downloading http://repo.maven.apache.org/maven2/com/google/api/grpc/proto-google-common-protos/1.0.0/proto-google-common-protos-1.0.0-sources.jar
11:39:05.709 [Worker-2] INFO  o.e.m.c.i.e.AbstractTransferListenerAdapter - Downloaded http://repo.maven.apache.org/maven2/com/google/api/grpc/proto-google-common-protos/1.0.0/proto-google-common-protos-1.0.0-sources.jar
11:39:06.069 [Worker-2] INFO  o.e.m.c.i.e.AbstractTransferListenerAdapter - Downloading http://repo.maven.apache.org/maven2/com/google/api/grpc/proto-google-iam-v1/0.1.24/proto-google-iam-v1-0.1.24-sources.jar
11:39:06.074 [Worker-2] INFO  o.e.m.c.i.e.AbstractTransferListenerAdapter - Downloaded http://repo.maven.apache.org/maven2/com/google/api/grpc/proto-google-iam-v1/0.1.24/proto-google-iam-v1-0.1.24-sources.jar
11:39:06.224 [Worker-8] INFO  o.e.m.c.i.e.AbstractTransferListenerAdapter - Downloading http://repo.maven.apache.org/maven2/com/google/api/api-common/1.2.0/api-common-1.2.0-sources.jar
11:39:06.229 [Worker-8] INFO  o.e.m.c.i.e.AbstractTransferListenerAdapter - Downloaded http://repo.maven.apache.org/maven2/com/google/api/api-common/1.2.0/api-common-1.2.0-sources.jar
11:39:06.403 [Worker-6] INFO  o.e.m.c.i.e.AbstractTransferListenerAdapter - Downloading http://repo.maven.apache.org/maven2/com/google/api/gax/1.14.0/gax-1.14.0-sources.jar
11:39:06.415 [Worker-6] INFO  o.e.m.c.i.e.AbstractTransferListenerAdapter - Downloaded http://repo.maven.apache.org/maven2/com/google/api/gax/1.14.0/gax-1.14.0-sources.jar
11:39:07.164 [Worker-7] INFO  o.e.m.c.i.e.AbstractTransferListenerAdapter - Downloading http://repo.maven.apache.org/maven2/com/google/cloud/google-cloud-core/1.10.0/google-cloud-core-1.10.0-sources.jar
11:39:07.170 [Worker-7] INFO  o.e.m.c.i.e.AbstractTransferListenerAdapter - Downloaded http://repo.maven.apache.org/maven2/com/google/cloud/google-cloud-core/1.10.0/google-cloud-core-1.10.0-sources.jar
11:39:07.290 [Worker-2] INFO  o.e.m.c.i.e.AbstractTransferListenerAdapter - Downloading http://repo.maven.apache.org/maven2/com/google/cloud/google-cloud-core-http/1.10.0/google-cloud-core-http-1.10.0-sources.jar
11:39:07.295 [Worker-2] INFO  o.e.m.c.i.e.AbstractTransferListenerAdapter - Downloaded http://repo.maven.apache.org/maven2/com/google/cloud/google-cloud-core-http/1.10.0/google-cloud-core-http-1.10.0-sources.jar
11:39:07.430 [Worker-11] INFO  o.e.m.c.i.e.AbstractTransferListenerAdapter - Downloading http://repo.maven.apache.org/maven2/com/google/cloud/google-cloud-datastore/1.10.0/google-cloud-datastore-1.10.0-sources.jar
11:39:07.437 [Worker-11] INFO  o.e.m.c.i.e.AbstractTransferListenerAdapter - Downloaded http://repo.maven.apache.org/maven2/com/google/cloud/google-cloud-datastore/1.10.0/google-cloud-datastore-1.10.0-sources.jar
Window manager warning: Buggy client sent a _NET_ACTIVE_WINDOW message with a timestamp of 0 for 0x400021 (data - Res)
Window manager warning: meta_window_activate called by a pager with a 0 timestamp; the pager needs to be fixed.
Window manager warning: Buggy client sent a _NET_ACTIVE_WINDOW message with a timestamp of 0 for 0x403b9c (Delete Res)
Window manager warning: meta_window_activate called by a pager with a 0 timestamp; the pager needs to be fixed.
11:39:16.347 [main] WARN  o.e.s.s.finder.widgets.SWTBotButton - Widget is not enabled: (of type 'Button' and with mnemonic 'OK' and with style 'SWT.PUSH')
11:39:16.347 [main] WARN  o.e.s.s.finder.widgets.SWTBotButton - Widget is not enabled: (of type 'Button' and with mnemonic 'OK' and with style 'SWT.PUSH')
11:39:16.347 [main] WARN  o.e.s.s.finder.widgets.SWTBotButton - Widget is not enabled: (of type 'Button' and with mnemonic 'OK' and with style 'SWT.PUSH')
11:39:16.347 [main] WARN  o.e.s.s.finder.widgets.SWTBotButton - Widget is not enabled: (of type 'Button' and with mnemonic 'OK' and with style 'SWT.PUSH')
11:39:16.347 [main] WARN  o.e.s.s.finder.widgets.SWTBotButton - Widget is not enabled: (of type 'Button' and with mnemonic 'OK' and with style 'SWT.PUSH')
Window manager warning: Buggy client sent a _NET_ACTIVE_WINDOW message with a timestamp of 0 for 0x400021 (data - Res)
Window manager warning: meta_window_activate called by a pager with a 0 timestamp; the pager needs to be fixed.
Tests run: 2, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 15.723 sec <<< FAILURE! - in com.google.cloud.tools.eclipse.integration.appengine.ImportNativeAppEngineStandardProjectTest
importAppEngineStandardJava7_from1_3_1(com.google.cloud.tools.eclipse.integration.appengine.ImportNativeAppEngineStandardProjectTest)  Time elapsed: 0.972 sec  <<< ERROR!
org.eclipse.swtbot.swt.finder.exceptions.AssertionFailedException: assertion failed: 
	at org.eclipse.swtbot.swt.finder.utils.internal.Assert.isTrue(Assert.java:95)
	at org.eclipse.swtbot.swt.finder.utils.internal.Assert.isTrue(Assert.java:83)
	at org.eclipse.swtbot.swt.finder.widgets.SWTBotTreeItem.select(SWTBotTreeItem.java:516)
	at org.eclipse.swtbot.swt.finder.widgets.SWTBotTreeItem.select(SWTBotTreeItem.java:543)
	at com.google.cloud.tools.eclipse.integration.appengine.SwtBotAppEngineActions.importNativeProject(SwtBotAppEngineActions.java:140)
	at com.google.cloud.tools.eclipse.integration.appengine.ImportNativeAppEngineStandardProjectTest.importAppEngineStandardJava7_from1_3_1(ImportNativeAppEngineStandardProjectTest.java:63)

importAppEngineStandardJava8_from1_3_1(com.google.cloud.tools.eclipse.integration.appengine.ImportNativeAppEngineStandardProjectTest)  Time elapsed: 14.73 sec
```

**Comment by chanseokoh** [comment]
The assertion failure is from `.select("Existing Projects into Workspace")`. Because `tree().expandNode("General")` is succeeding, the node exists but it's likely that it has not yet completed loading all wizards under it.

**Comment by chanseokoh** [comment]
#2598 still doesn't fix it. The root problem seems to be that the wizard we are looking for ("Existing Projects into Workspace") never appears, or the SWT bot isn't able to find it for some reason.

(BTW, the error message should read "Tree item never appeared" instead of "Tree items never disappeared".)
```
[Watchdog] > importAppEngineStandardJava8_from1_3_1(com.google.cloud.tools.eclipse.integration.appengine.ImportNativeAppEngineStandardProjectTest)
Tests run: 2, Failures: 0, Errors: 1, Skipped: 1, Time elapsed: 48.599 sec <<< FAILURE! - in com.google.cloud.tools.eclipse.integration.appengine.ImportNativeAppEngineStandardProjectTest
importAppEngineStandardJava7_from1_3_1(com.google.cloud.tools.eclipse.integration.appengine.ImportNativeAppEngineStandardProjectTest)  Time elapsed: 34.548 sec  <<< ERROR!
org.eclipse.swtbot.swt.finder.widgets.TimeoutException: Timeout after: 30000 ms.: Tree items never disappeared
	at org.eclipse.swtbot.swt.finder.SWTBotFactory.waitUntil(SWTBotFactory.java:437)
	at org.eclipse.swtbot.swt.finder.SWTBotFactory.waitUntil(SWTBotFactory.java:411)
	at org.eclipse.swtbot.swt.finder.SWTBotFactory.waitUntil(SWTBotFactory.java:399)
	at com.google.cloud.tools.eclipse.swtbot.SwtBotTreeUtilities.waitUntilTreeItemhasChild(SwtBotTreeUtilities.java:369)
	at com.google.cloud.tools.eclipse.integration.appengine.SwtBotAppEngineActions.importNativeProject(SwtBotAppEngineActions.java:142)
	at com.google.cloud.tools.eclipse.integration.appengine.ImportNativeAppEngineStandardProjectTest.importAppEngineStandardJava7_from1_3_1(ImportNativeAppEngineStandardProjectTest.java:63)

importAppEngineStandardJava8_from1_3_1(com.google.cloud.tools.eclipse.integration.appengine.ImportNativeAppEngineStandardProjectTest) skipped
```

**Comment by chanseokoh** [comment]
I triggered a lot of tests with #2603, and the logs shows that the "General" node is expanded but has no child.

```
TreeItem with text {General}: expanded? true
    TreeItem with text {}
Window manager warning: Buggy client sent a _NET_ACTIVE_WINDOW message with a timestamp of 0 for 0x400021 (Resource -)
Window manager warning: meta_window_activate called by a pager with a 0 timestamp; the pager needs to be fixed.
[Watchdog] > importAppEngineStandardJava8_from1_3_1(com.google.cloud.tools.eclipse.integration.appengine.ImportNativeAppEngineStandardProjectTest)
Tests run: 2, Failures: 0, Errors: 1, Skipped: 1, Time elapsed: 36.214 sec <<< FAILURE! - in com.google.cloud.tools.eclipse.integration.appengine.ImportNativeAppEngineStandardProjectTest
importAppEngineStandardJava7_from1_3_1(com.google.cloud.tools.eclipse.integration.appengine.ImportNativeAppEngineStandardProjectTest)  Time elapsed: 32.145 sec  <<< ERROR!
org.eclipse.swtbot.swt.finder.widgets.TimeoutException: Timeout after: 30000 ms.: Tree item never appeared
	at org.eclipse.swtbot.swt.finder.SWTBotFactory.waitUntil(SWTBotFactory.java:437)
	at org.eclipse.swtbot.swt.finder.SWTBotFactory.waitUntil(SWTBotFactory.java:411)
	at org.eclipse.swtbot.swt.finder.SWTBotFactory.waitUntil(SWTBotFactory.java:399)
	at com.google.cloud.tools.eclipse.swtbot.SwtBotTreeUtilities.waitUntilTreeItemhasChild(SwtBotTreeUtilities.java:369)
	at com.google.cloud.tools.eclipse.integration.appengine.SwtBotAppEngineActions.importNativeProject(SwtBotAppEngineActions.java:143)
	at com.google.cloud.tools.eclipse.integration.appengine.ImportNativeAppEngineStandardProjectTest.importAppEngineStandardJava7_from1_3_1(ImportNativeAppEngineStandardProjectTest.java:63)
```

**Comment by chanseokoh** [comment]
I get a feeling that #2603 will prevent this. Let me close this, but I will re-open if this ever happens again.

**Comment by briandealwis** [comment]
I just saw this happen!  I don't know why, but a workaround would seem to be to collapse and then re-expand the tree item.

![screen shot 2017-12-13 at 3 54 10 pm](https://user-images.githubusercontent.com/202851/33962250-e97d138e-e01e-11e7-92ea-c975b00512be.PNG)

**Comment by chanseokoh** [comment]
Did your build have #2603 in it? #2603 should have closed the dialog and re-open after timeout.

**Comment by chanseokoh** [comment]
Well, I guess you had #2603 because I can see the diagnostic message. Also looks like the fix of #2603 didn't help, as I see the stack dump log after 2 minutes.

**Comment by briandealwis** [comment]
I did, but I had a breakpoint on `TimeoutException` and so caught it in the act.  I just reproduced it on a second run through.  I'll see if I can figure out why this happens.
